contracts/Content/UniqueContent.sol

```diff
@@ -48,18 +53,13 @@ contract UniqueContent is IUniqueContent, IMultipleRoyalties, MultipleRoyalties,
     */
     function burn(uint256 _uniqueId) external override {
         require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to burn it");
-        require(uniqueAssetInfo[_uniqueId].creator == _msgSender() || !uniqueAssetInfo[_uniqueId].creatorLocked, "Creator has disabled the burning of this token");
+        require(uniqueContentStorage.isCreator(_uniqueId, _msgSender()) || !uniqueContentStorage.isLocked(_uniqueId), "Creator has disabled the burning of this token");
         _burn(_uniqueId);
         // transfers original asset back to caller
-        IERC1155Upgradeable(uniqueAssetInfo[_uniqueId].contentAddress).safeTransferFrom(address(this), _msgSender(), uniqueAssetInfo[_uniqueId].tokenId, 1, "");
+        (uint256 _tokenId, address _contentAddress) = uniqueContentStorage.getAssetData(_uniqueId);
+        IERC1155Upgradeable(_contentAddress).safeTransferFrom(address(this), _msgSender(), _tokenId, 1, "");
```

Comment:
Now that the asset is stored in a storage, you can now switch this. Call burnUniqueAssetInfo() first before the safeTransferFrom() call. "transfer" before "update storage data" can cause a re-entrancy issue. 

Do: 
// to store it into a local variable
(uint256 _tokenId, address _contentAddress) = uniqueContentStorage.getAssetData(_uniqueId);

// burn 
uniqueContentStorage.burnUniqueAssetInfo(_uniqueId);

// transfer
IERC1155Upgradeable(_contentAddress).safeTransferFrom(address(this), _msgSender(), _tokenId, 1, "");